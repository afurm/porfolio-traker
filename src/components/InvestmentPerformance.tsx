import React from 'react';
import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer } from 'recharts';
import { Box, Typography, Paper } from '@mui/material';

export interface InvestmentPerformanceProps {
  data: { month: string; profit: number; loss: number }[];
}

const InvestmentPerformance: React.FC<InvestmentPerformanceProps> = ({ data }) => {
  return (
    <Box sx={{ p: 4, maxWidth: 800, mx: 'auto' }}>
      <Paper elevation={3} sx={{ p: 3, borderRadius: 2 }}>
        <Typography variant="h5" fontWeight="bold" textAlign="center" mb={3}>
          Investment Performance
        </Typography>
        <ResponsiveContainer width="100%" height={400}>
          <BarChart data={data} margin={{ top: 20, right: 30, left: 20, bottom: 5 }}>
            <CartesianGrid strokeDasharray="3 3" />
            <XAxis dataKey="month" />
            <YAxis />
            <Tooltip />
            <Legend />
            <Bar dataKey="profit" fill="#4caf50" />
            <Bar dataKey="loss" fill="#f44336" />
          </BarChart>
        </ResponsiveContainer>
      </Paper>
    </Box>
  );
};

export default InvestmentPerformance;