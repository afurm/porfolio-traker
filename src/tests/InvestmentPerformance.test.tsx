import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import InvestmentPerformance, { InvestmentPerformanceProps } from '@/components/InvestmentPerformance';

global.ResizeObserver = class {
  observe() {}
  unobserve() {}
  disconnect() {}
};

// Sample data for the tests
const testData: InvestmentPerformanceProps['data'] = [
  { month: 'January', profit: 5000, loss: 1000 },
  { month: 'February', profit: 3000, loss: 2000 },
  { month: 'March', profit: 7000, loss: 1500 },
];

describe('InvestmentPerformance Component', () => {
  // Snapshot Test
  it('matches snapshot', () => {
    const { asFragment } = render(<InvestmentPerformance data={testData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Render Tests
  it('renders Investment Performance title', () => {
    render(<InvestmentPerformance data={testData} />);
    expect(screen.getByText('Investment Performance')).toBeInTheDocument();
  });
});
