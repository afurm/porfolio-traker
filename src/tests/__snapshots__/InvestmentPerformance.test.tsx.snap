// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestmentPerformance Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s37v5s"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-yt9ce0-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ffs2s2-MuiTypography-root"
      >
        Investment Performance
      </h5>
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 400px; min-width: 0;"
      />
    </div>
  </div>
</DocumentFragment>
`;
